src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -139,17 +397,13 @@ private static async Task<VersionStamp> GetDocumentSemanticVersionAsync(Document
             // We return null (which to the LSP is a no-op) to prevent flashing of CSharp elements.
             if (combinedSemanticRanges is null)
             {
-                return null;
+                return (null, isCSharpFinalized);
             }
 
             var data = ConvertSemanticRangesToSemanticTokensData(combinedSemanticRanges, codeDocument);
             var tokens = new SemanticTokens { Data = data };
 
-            // Cache the result so we can potentially avoid recomputation next time around.
-            var cacheResponse = new SemanticTokensCacheResponse(semanticVersion, range, tokens, isCSharpFinalized);
-            _cachedResponses.Set(textDocumentIdentifier.Uri, cacheResponse);
-
-            return tokens;
+            return (tokens, isCSharpFinalized);
```

Comment:
instead of returning this vlaue out so that the caller can decide to not cache, why not do that right here?

Reply:
There are some situations where we don't want to cache, for example if the LSP client sends us a request containing a partial line (can happen if the user is at the end of a document which doesn't end with a newline).

Reply:
Actually, after looking again I think you're right that the returned value is unnecessary - will move to the inner method